packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -102,10 +136,163 @@ class CupertinoContextMenu extends StatefulWidget {
+  /// and when it is fully open. This will overwrite the default animation that
+  /// matches the behavior of an iOS 16.0 context menu.
+  ///
+  /// This builder is used instead of the child when either the intended child
+  /// has a propert that would conflict with the default animation, like a
+  /// border radius or a shadow, or if simply a more custom animation is needed.
+  ///
+  /// In addition to the current [BuildContext], the function is also called
+  /// with an [Animation]. The animation goes from 0 to 1 when
+  /// the CupertinoContextMenu opens, and from 1 to 0 when it closes, and it can
+  /// be used to animate the widget in sync with this opening and closing.
+  ///
+  /// {@tool snippet}
+  ///
+  /// Below is an example of using [builder] to show an image tile setup to be
+  /// opened in the default way to match a native iOS 16.0 app. This can be
+  /// manipulated to add more custamizability to the widget's animation.
+  ///
+  /// ```dart
+  /// CupertinoContextMenu(
+  ///   actions: <Widget>[
+  ///     CupertinoContextMenuAction(
+  ///       child: const Text('Action one'),
+  ///       onPressed: () {},
+  ///     ),
+  ///   ],
+  ///   builder:(BuildContext context, Animation<double> animation) {
+  ///     final Animation<BorderRadius?> borderRadiusAnimation = BorderRadiusTween(
+  ///       begin: BorderRadius.circular(0.0),
+  ///       end: BorderRadius.circular(CupertinoContextMenu.kOpenBorderRadius),
+  ///     ).animate(
+  ///       CurvedAnimation(
+  ///         parent: animation,
+  ///         curve: Interval(
+  ///           CupertinoContextMenu.animationOpensAt,
+  ///           1.0,
+  ///         ),
+  ///       ),
+  ///      );
+  ///
+  ///     final Animation<Decoration> boxDecorationAnimation = DecorationTween(
+  ///       begin: const BoxDecoration(
+  ///        color: Color(0xFFFFFFFF),
+  ///        boxShadow: <BoxShadow>[],
+  ///       ),
+  ///       end: const BoxDecoration(
+  ///        color: Color(0xFFFFFFFF),
+  ///        boxShadow: CupertinoContextMenu.kEndBoxShadow,
+  ///       ),
+  ///      ).animate(
+  ///        CurvedAnimation(
+  ///         parent: animation,
+  ///         curve: Interval(
+  ///           0.0,
+  ///           CupertinoContextMenu.animationOpensAt,
+  ///       ))
+  ///     );
+  ///
+  ///     return Container(
+  ///       decoration:
+  ///         animation.value < CupertinoContextMenu.animationOpensAt ? boxDecorationAnimation.value : null,
+  ///       child: FittedBox(
+  ///         fit: BoxFit.cover,
+  ///         child: ClipRRect(
+  ///           borderRadius: borderRadiusAnimation.value ?? BorderRadius.circular(0.0),
+  ///           child: SizedBox(
+  ///             height: 150,
+  ///             width: 150,
+  ///             child: Image.network('https://picsum.photos/250?image=9'),
+  ///           ),
+  ///         ),
+  ///       )
+  ///     );
+  ///   },
+  /// )
+  /// ```
```

Comment:
This is a great example that shows everything a user would need to do to get this to look perfect for their own widgets, but I have one nitpick.  In the case of a rectangular image with no transparency like this, wouldn't it work just as well to simply pass it as the `child`?  Is there something else we could use to demonstrate the need for builder without making it super complicated?

If not, maybe at least mention that this is identical to just using `child`.

```dart
CupertinoContextMenu(
  actions: <Widget>[
    CupertinoContextMenuAction(
      child: const Text('Action one'),
      onPressed: () {},
    ),
  ],
  child: SizedBox(
    height: 150,
    width: 150,
    child: Image.network('https://picsum.photos/250?image=9'),
  ),
),
```

Reply:
Also, it's probably worth it to put this example in the `examples/` directory and test it.  Let me know if you're not familiar with how to do that.

Reply:
Ah I see, that makes sense.  It would be great to have another separate example of something that needs a `builder` if you have time to add it.

I didn't realize there was an existing example for CupertinoContextMenu but you're [right](https://github.com/flutter/flutter/blob/master/examples/api/lib/cupertino/context_menu/cupertino_context_menu.0.dart).  If that one still looks ok with your updates then I say just keep it as-is and add a separate one that uses `builder`.  You can just copy it, its [test](https://github.com/flutter/flutter/blob/master/examples/api/test/cupertino/context_menu/cupertino_context_menu.0_test.dart), and its [reference](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/cupertino/context_menu.dart#L86) in the docs and go from there and it should be pretty straightforward.